src/PerforceService.ts

```diff
@@ -240,7 +232,13 @@ export namespace PerforceService {
             allArgs.push(...args);
         }
 
-        const spawnArgs: CP.SpawnOptions = { cwd: config ? config.localDir : wksPath };
+        const cwd = config
```

Comment:
Note to self- use `??` Or `||` here instead

Reply:
In 13 years of using perforce I never needs the -d global option on the command line. Original code seems to indicate it was because of multi roots in the vscode workspace but I don’t understand that either. Are workspaces and the scm client related somehow?

Reply:
Because perforce may infer the client, user, port settings etc from the directory where you run the command, if the directory or one of its parents contains a p4config file. 

So if you open a file in vscode from outside of your vscode workspace, and that file is under a dir with a p4config, by running in the file's directory we can still use perform actions on the file